modules/oe_theme_helper/src/Plugin/field_group/FieldGroupFormatter/FieldInPageNavigationFormatter.php

```diff
@@ -0,0 +1,201 @@
+
+  /**
+   * {@inheritdoc}
+   */
+  public function process(&$element, $processed_object) {
+    $element += [
+      '#type' => 'field_group_in_page_navigation',
+    ];
+    // Get children elements in in page navigation group.
+    $field_elements = $processed_object['#fieldgroups'][$element['#group_name']]->children;
+    // Add anchors links and anchors to children with labels.
+    $links = [];
+    $i = 1;
+    foreach ($field_elements as $key => $value) {
+      // Check if we should add the field to page contents.
+      if (!is_array($element[$value]) || $this->excludeField($value, $element[$value], $processed_object['#fieldgroups']) === TRUE) {
+        continue;
+      }
+      // We will add anchors when the element, field or group, has a label.
+      $label = $this->fieldGetLabel($value, $processed_object);
+      // If there is no label, don't add if to the menu.
+      if ($label === '') {
+        continue;
+      }
+      // If a group has a default id, use it as anchor link.
+      $type = $this->groupOrField($value, $processed_object['#fieldgroups']);
+      if ($type === 'group' && isset($processed_object['#fieldgroups'][$value]->format_settings['id']) && $processed_object['#fieldgroups'][$value]->format_settings['id'] !== '') {
+        $id = $processed_object['#fieldgroups'][$value]->format_settings['id'];
+      }
+      else {
+        $id = "inline-nav-" . $i++;
+      }
+      $links[] = [
+        'href' => "#" . $id,
+        'label' => $label,
+      ];
+      $element[$value]['#id'] = $id;
+      $element[$value]['#attributes']['id'] = $id;
+    }
+    $element['#links'] = $links;
+    if (count($links) > 0) {
+      $element['#inline_title'] = t('Page contents');
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function preRender(&$element, $rendering_object) {
+    parent::preRender($element, $rendering_object);
+    $this->process($element, $rendering_object);
+  }
+
+  /**
+   * Check if the field should create an in page navigation entry.
+   *
+   * @param string $field_name
+   *   Name of the field.
+   * @param array $field
+   *   The field to check.
+   * @param array $fieldgroups
+   *   List of groups.
+   *
+   * @return bool
+   *   Include or not the field.
+   */
+  private function excludeField(string $field_name, array $field, array $fieldgroups) :bool {
+    $type = $this->groupOrField($field_name, $fieldgroups);
+    switch ($type) {
+      case "field":
+        // Exclude empty fields.
+        if ($type === 'field' && !isset($field['#items'])) {
+          return TRUE;
+        }
+
+        break;
+
+      case 'group':
+        // Exclude empty groups.
+        if (count($field) < 1) {
+          return TRUE;
+        }
+        foreach ($field as $value) {
+          // If there is a field with values, don't exclude the group.
+          if (isset($value['#items'])) {
+            return FALSE;
+          }
+        }
```

Comment:
Empty line would be nice after the foreach statement here.

Reply:
Add line